File: ios/Snacks/Snacks/src/OArray.swift
//
//  OArray.swift
//  snacks
//
//  Created by noear on 16/2/21.
//  Copyright © 2016年 noear. All rights reserved.
//

import Foundation

public class OArray{
    public final var elements:[ONode] = [];
    
    public func add(_ value:ONode){
        elements.append(value);
    }
    
    public func add(_ index:Int,_ value:ONode){
        elements.insert(value, at: index);
    }
    
    public func get(_ index:Int) -> ONode{
        return elements[index];
    }
    
    public func count()->Int{
        return elements.count;
    }
    
    public func clear(){
        elements.removeAll();
    }
}

File: java/Snacks/src/main/java/org/noear/snacks/OArray.java
package org.noear.snacks;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by noear on 14-6-11.
 */
public class OArray {
    public final List<ONode> elements;

    public  OArray(){
         elements = new ArrayList<ONode>();
    }

    public void add(ONode value)
    {
        elements.add(value);
    }

    public void add(int index, ONode value)
    {
        elements.add(index,value);
    }

    public ONode get(int index)
    {
        return elements.get(index);
    }

    public void removeAt(int index){
        elements.remove(index);
    }

    public int count(){
        return elements.size();
    }

    public void clear(){
        elements.clear();
    }
}
